x-pack/plugins/observability/public/components/custom_threshold/custom_threshold_rule_expression.tsx

```diff
@@ -96,14 +100,24 @@ export default function Expressions(props: Props) {
       let initialSearchConfiguration = ruleParams.searchConfiguration;
 
       if (!ruleParams.searchConfiguration || !ruleParams.searchConfiguration.index) {
-        const newSearchSource = data.search.searchSource.createEmpty();
-        newSearchSource.setField('query', data.query.queryString.getDefaultQuery());
-        const defaultDataView = await data.dataViews.getDefaultDataView();
-        if (defaultDataView) {
-          newSearchSource.setField('index', defaultDataView);
-          setDataView(defaultDataView);
+        if (metadata?.currentOptions?.searchConfiguration) {
```

Comment:
There are many nested IFs here, can you simplify the flow?

Reply:
e.g., making one of the conditions as default value/path (the one that most likely will happen).

Reply:
Humm, the challenge here is that we have different sources for `searchConfiguration`, it can be in the edit mode, so we already have it, it might be from currentOptions, or we might not have it at all, so we want to use the default data view then. And we have both state and set it in rule params. Still, if you know how to improve it, please let me know.